Drop unused dependency on sequencer

henriquesimoes opened the pull request:
Since 35ef945 (Applied patch to update Makefiles, RELEASE and CONFIG_SITE files, 2019-04-17), parkerIOC is specified to link sequencer libraries (that is, seq and pv), but it actually doesn't contain any sequencer program. Drop this dependency to simplify the IOC settings, and its build process.

---

I don't get why it was added in the first place, so I might be missing something. Let me know if that's the case.

In addition, I don't have experience with setting up those CI files (both EPICS CI and Travis CI), so I wouldn't be surprised if I dropped something that needs to be around or if I didn't drop something useless. Please, let me know as well so that I can rewrite the commit.

henriquesimoes committed:
Drop unused dependency on sequencer

Since 35ef945 (Applied patch to update Makefiles, RELEASE and
CONFIG_SITE files, 2019-04-17), parkerIOC is specified to link sequencer
libraries (that is, seq and pv), but it actually doesn't contain any
sequencer program. Drop this dependency to simplify the IOC settings,
and its build process.

ericonr commented on iocs/parkerIOC/parkerApp/src/Makefile:
```diff
@@ -33,9 +33,6 @@ parker_LIBS += motor
 #ifdef ASYN
 parker_LIBS += asyn
 #endif
-#ifdef SNCSEQ
-parker_LIBS += seq pv
-#endif
```
Since we are here, could you add a commit before this one removing all the `#ifdef` and `#endif` lines? They are misleading, because they are commented out but they look like C preprocessor directives.

henriquesimoes replied:
Done. Thanks! That was something that misled me at first as well.

henriquesimoes committed:
Drop commented out ifdef conditionals

Having ifdef lines commented out in Makefile syntax makes them look like
their C preprocessor counterparts. Drop them altogether to avoid to
misleading readers to think those dependencies are conditionally
defined.

Suggested-by: Érico Nogueira Rolim <erico.rolim@lnls.br>

kmpeters commented:
It looks like the patch file I used when I created the example IOC after splitting the motor module into many driver submodules included the commented out ifdef lines:

```Makefile
#ifdef SNCSEQ
parker_LIBS += seq pv
#endif
```

Instead of this, which was the intent (and would not require the sequencer to build):

```Makefile
ifdef SNCSEQ
parker_LIBS += seq pv
endif
```

henriquesimoes commented:
> Instead of this, which was the intent (and would not require the sequencer to build)

I see. Which path forward do you want to follow?

In my humble opinion, it is better not to have it listed at all (not even with an actual `ifdef` conditional). After all, there is no sequencer program in this specific IOC to be optionally built. It gives the false impression that it has by having such a `ifdef` in the `Makefile`.

If it ever gets added, we can add  those lines back to where they were.

kmpeters commented:
@henriquesimoes, I don't have a preference between the two approaches.  Removing the sequencer lines is ok with me.  I only care that the example IOC builds and runs.

kmpeters committed:
Merge branch 'master' into drop-seq-deps

ericonr commented:
Kevin, Henrique is enjoying his holidays, so he will be unable to respond for a while. If you are okay with the approach taken in this PR, we would be interested in having it merged :)

kmpeters commented:
@henriquesimoes, is this pull req ready to merge?

ericonr commented:
Cheers, thank you!
